tests/spec/s-function.js

```diff
@@ -1,7 +1,177 @@
 /* global describe, it, expect, beforeEach */
+var hasStrictMode = (function () {
+  'use strict';
+
+   return !this;
+}());
+
+var global = (function () {
+   return this;
+}());
 
 describe('Function', function () {
-    'use strict';
+    describe('#call()', function () {
+        var noop = function () {};
+        it('should not throw a error in each case', function () {
+            expect(function () {
+                noop.call();
+            }).not.toThrow();
+            expect(function () {
+                /*eslint no-useless-call: 0*/
+                noop.call(undefined);
+            }).not.toThrow();
+            expect(function () {
+                /*eslint no-useless-call: 0*/
+                noop.call(null);
+            }).not.toThrow();
+            expect(function () {
+                noop.call('a');
+            }).not.toThrow();
+            expect(function () {
+                noop.call(1);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(true);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(new Date());
+            }).not.toThrow();
+            expect(function () {
+                noop.call(/test/);
+            }).not.toThrow();
+        });
+        it('should not throw a error with additional args', function () {
+            expect(function () {
+                noop.call(undefined, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(null, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call('a', undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(1, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(true, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(noop, undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(new Date(), undefined, null, '1', 1, true, noop);
+            }).not.toThrow();
+        });
+        if (hasStrictMode) {
```

Comment:
instead of this if-wrapping, please use the approach I've used in most other test files - something like:
```js
var ifHasStrictIt = hasStrictMode ? it : xit;`
```

(I'll fix the current instance of this pattern in the array tests)